shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/util/PipelineJobPreparerUtils.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.shardingsphere.data.pipeline.core.prepare.datasource.PrepareTargetSchemasParameter;
+import org.apache.shardingsphere.data.pipeline.core.prepare.datasource.PrepareTargetTablesParameter;
+import org.apache.shardingsphere.data.pipeline.spi.check.datasource.DataSourceChecker;
+import org.apache.shardingsphere.data.pipeline.spi.check.datasource.DataSourceCheckerFactory;
+import org.apache.shardingsphere.data.pipeline.spi.ingest.position.PositionInitializer;
+import org.apache.shardingsphere.infra.database.type.DatabaseType;
+import org.apache.shardingsphere.infra.datasource.pool.creator.DataSourcePoolCreator;
+import org.apache.shardingsphere.infra.datasource.props.DataSourceProperties;
+import org.apache.shardingsphere.infra.yaml.config.swapper.resource.YamlDataSourceConfigurationSwapper;
+
+import javax.sql.DataSource;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Pipeline job preparer utils.
+ */
+@Slf4j
+public final class PipelineJobPreparerUtils {
+    
+    /**
+     * Prepare target schema.
+     *
+     * @param databaseType database type
+     * @param prepareTargetSchemasParameter prepare target schemas parameter
+     */
+    public static void prepareTargetSchema(final String databaseType, final PrepareTargetSchemasParameter prepareTargetSchemasParameter) {
+        Optional<DataSourcePreparer> dataSourcePreparer = DataSourcePreparerFactory.getInstance(databaseType);
+        if (!dataSourcePreparer.isPresent()) {
+            log.info("dataSourcePreparer null, ignore prepare target");
+            return;
+        }
+        dataSourcePreparer.get().prepareTargetSchemas(prepareTargetSchemasParameter);
+    }
+    
+    /**
+     * Prepare target tables.
+     *
+     * @param databaseType database type
+     * @param prepareTargetTablesParameter prepare target tables parameter
+     */
+    public static void prepareTargetTables(final String databaseType, final PrepareTargetTablesParameter prepareTargetTablesParameter) {
+        Optional<DataSourcePreparer> dataSourcePreparer = DataSourcePreparerFactory.getInstance(databaseType);
+        if (!dataSourcePreparer.isPresent()) {
+            log.info("dataSourcePreparer null, ignore prepare target");
+            return;
+        }
+        dataSourcePreparer.get().prepareTargetTables(prepareTargetTablesParameter);
+    }
+    
+    /**
+     * Get incremental position.
+     *
+     * @param initProgress init progress
+     * @param dumperConfig dumper config
+     * @param dataSourceManager data source manager
+     * @return ingest position
+     * @throws SQLException sql exception
+     */
+    public static IngestPosition<?> getIncrementalPosition(final JobProgress initProgress, final DumperConfiguration dumperConfig,
+                                                           final PipelineDataSourceManager dataSourceManager) throws SQLException {
```

Comment:
Could we just transfer `JobItemIncrementalTasksProgress initIncremental` to replace `JobProgress initProgress`?